### PR TITLE
[confmap] Move confmap.unifyEnvVarExpansion to beta

### PR DESCRIPTION
**Pull request by TylerHelmuth**
#### Description
Moves confmap.unifyEnvVarExpansion to beta.  This means the collector will, by default, use the env var provider to expand `${FOO}` synatx and will error if the expandconverter is used to expand `$FOO` syntax.

#### Link to tracking issue
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/10161
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/8215
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/7111

### CODE REVIEW

**Comment by TylerHelmuth** [comment]
We'll want to make a big deal in this release notes/changelog about this breaking change.

**Comment by mx-psi** [comment]
Agreed about the changelog. I would alsp like to check our docs and examples to remove any references to the `$ENV` syntax. It would also help to have a very clear error message with instructions if you run into this issue

**Comment by mx-psi** [comment]
I searched on several repositories using `rg '\$[A-Z]' -tyaml --glob='!.github/'`:

- opentelemetry-collector: No relevant hits
- opentelemetry-collector-contrib: No relevant hits (Two files on `exporter/splunkhecexporter/`
- opentelemetry.io: No relevant hits

**Comment by TylerHelmuth** [comment]
> It would also help to have a very clear error message with instructions if you run into this issue

Updated the error message to resemble the warning log, but with details about the feature gate.

**Comment by TylerHelmuth** [comment]
I confirmed that opentelemetry-helm-charts and opentelemetry-operator have no relevant hits

File: confmap/converter/expandconverter/expand.go
Comment (mx-psi) [inline confmap/converter/expandconverter/expand.go:97]: Should we make it clear that this is a temporary solution and that you have to migrate?
```suggestion
				err = fmt.Errorf("variable substitution using $VAR has been deprecated in favor of ${VAR} and ${env:VAR} - please update $%s or temporarily disable the confmap.unifyEnvVarExpansion feature gate", str)
```

**Review by mx-psi — APPROVED** [review]
LGTM. This is likely to impact some users, so I would like to have more reviews from other approvers before merging

**Comment by mx-psi** [comment]
cc @open-telemetry/collector-approvers I would like more approvals here before merging

**Review by songy23 — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Comment by azunna1** [comment]
This change is breaking my prometheus receiver config where i have relabelling like
```
relabel_configs:
                - source_labels: [job]
                  separator: ;
                  regex: (.*)
                  target_label: __tmp_prometheus_job_name
                  replacement: $$1
                  action: replace
```
How can we now put config variables without otel treating it like an environment variable?

**Comment by TylerHelmuth** [comment]
@azunna1 I am unable to reproduce your issue, can you open an issue with more details.

**Comment by azunna1** [comment]
@TylerHelmuth Please Ignore that. The issue was coming from regex substitution config. For example
```
metricstransform:
        transforms:
          - include: '^k8s\.(.*)\.(.*)$$'
            match_type: regexp
            action: update
            new_name: 'kubernetes.$${1}.$${2}'
          - include: '^container_([0-9A-Za-z]+)_([0-9A-Za-z]+)_.*'
            match_type: regexp
            action: update
            new_name: 'container.$${1}.$${2}'
```
https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/metricstransformprocessor#rename-multiple-metrics-using-substitution

Please let me know if its an error from my syntax. If its a bug, i can open anew issue, thanks

**Pull request merged**
